com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/bundle/resources/jsweb/dmloader.js

```diff
@@ -499,6 +544,50 @@ var Progress = {
     }
 };
 
+var ProgressUpdater = {
```

Comment:
Introduce new object `ProgressUpdater`. It should control and recalculate downloading progress and notify all listeners.